### DISCUSSION TITLE
Solr logging suddenly stops and breaks whole docker daemon

### DISCUSSION THREAD

**Discussion by Phil1108** [docker-solr/docker-solr#414] [Q&A]
At a certain point (coincidental seems to be at the same time where the commits happen), the Solr container is not printing out any logs (except the Prometheus exporter logs). Also when I go into the container the log file itself is not updated. You can still request Solr most (not all) of the time without any error.  

If I now want to restart the Solr docker container, the whole docker daemon becomes unresponsive (Running in Timeouts)  with this error: 
`ERROR: Couldn't connect to Docker daemon at http+docker://localhost - is it running?`

The only way to recover is to restart the whole docker daemon: 
`sudo systemctl restart docker`

I first suspected that this is a Disk IO Problem, so I switched the index core to an NVMe SSD and it still occured.
Also I couldn't detect any swap usage of the docker container, but I am not quite sure how I can be sure that the JVM is not shifting anything to the swap. 

Any chance how I can further debug it? It is extremely hard without any solr logs to specifiy the problem more. Any chance that I can get more logs about the JVM in general? 

Memory:
- Xms5g -Xmx5g
- Docker container started with `mem_limit: 15000m`
- System with 32 GB (No OOM visible)

Solr Config: 
autoCommit.maxTime=30s
opensearcher=false
autoSoftCommit=300s

I have the following setup: 
- docker 20.10.12
- solr docker-container 8.6.3 (also tested with 8.11.1)
- Only a single core with no replication or sharding: The core is quite small (1000 Documents, Less than 5 mb in total core size)

**Comment by dsmiley** [comment]
Weird!  This is suggestive of some sort of host/env issue but who knows.  So the log files aren't being written to anymore?  And the log files are in a volume, perhaps mounted to the host?  If not, I'd do that.  You might also consider configuring Log4j2 to log to something else that isn't disk based for the purpose of this investigation.  If it is something with this machine/host, you might try reproducing it on some other machine/setup.  Ideally docker (container tech generally) would make such issues rare.

**Reply by Phil1108** [reply]
We recreated it on multiple machines with different setups of disks (hdd vs ssd vs m2) but nothing really helped. 
By accident we found out that at least until know it has never appeared when we switched to podman. 
So might indeed be a docker bug and only indirectly connected to solr docker images
